### PR TITLE
fix: hovered menubutton icon does not disappear

### PR DESCRIPTION
**Pull request by smhigley**
## Previous Behavior

We had a stray `:hover` style that made icons invisible on hover in expanded menubuttons:
Screenshot of an expanded menubutton with menu in dark high contrast mode, and there is an empty dark space to the left of the light blue button text

## New Behavior

Uses the correct color token (Highlight) for the color, which matches the color/bg color pair used in Button's hover styles

## Related Issue(s)

Fixes an issue found while debugging icon high contrast color issues

### CODE REVIEW

**Review by tomi-msft — APPROVED** [review]

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports | Baseline (minified/GZIP) | PR    | Change     |
| :---------------- | -----------------------: | ----: | ---------: |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/MenuButton.fixture.js'>MenuButton</abbr> | `41.916 kB`<br />`12.119 kB` | `41.919 kB`<br />`12.108 kB` | `3 B` <br />`-11 B` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/SplitButton.fixture.js'>SplitButton</abbr> | `49.933 kB`<br />`13.693 kB` | `49.936 kB`<br />`13.681 kB` | `3 B` <br />`-12 B` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ReactComponents.fixture.js'>react-components: entire library</abbr> | `1.1 MB`<br />`272.015 kB` | `1.1 MB`<br />`272.006 kB` | `3 B` <br />`-9 B` |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports | Size (minified/GZIP) |
| ----------------- | -------------------: |
| <samp>react-breadcrumb</samp> <br /> <abbr title='bundle-size/index.fixture.js'>@fluentui/react-breadcrumb - package</abbr> | `113.699 kB`<br />`31.498 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/Button.fixture.js'>Button</abbr> | `37.104 kB`<br />`10.766 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/CompoundButton.fixture.js'>CompoundButton</abbr> | `43.518 kB`<br />`12.068 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/ToggleButton.fixture.js'>ToggleButton</abbr> | `53.037 kB`<br />`12.528 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/Card.All.fixture.js'>Card - All</abbr> | `101.248 kB`<br />`28.626 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/Card.fixture.js'>Card</abbr> | `94.022 kB`<br />`26.806 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardFooter.fixture.js'>CardFooter</abbr> | `14.355 kB`<br />`5.79 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardHeader.fixture.js'>CardHeader</abbr> | `16.888 kB`<br />`6.669 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardPreview.fixture.js'>CardPreview</abbr> | `14.42 kB`<br />`5.922 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr> | `69.14 kB`<br />`20.137 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `218.317 kB`<br />`63.258 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr> | `44.447 kB`<br />`14.59 kB` |
| <samp>react-dialog</samp> <br /> <abbr title='bundle-size/Dialog.fixture.js'>Dialog (including children components)</abbr> | `99.724 kB`<br />`29.912 kB` |
| <samp>react-message-bar</samp> <br /> <abbr title='bundle-size/Default.fixture.js'>MessageBar (all components)</abbr> | `24.722 kB`<br />`9.205 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr> | `8.39 kB`<br />`2.64 kB` |
| <samp>react-tag-picker</samp> <br /> <abbr title='bundle-size/Package.fixture.js'>@fluentui/react-tag-picker - package</abbr> | `183.584 kB`<br />`55.264 kB` |
| <samp>react-timepicker-compat</samp> <br /> <abbr title='bundle-size/TimePicker.fixture.js'>TimePicker</abbr> | `107.387 kB`<br />`35.758 kB` |
</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/12d545e2f9eda489784fc1c4869a70fd50a4f5c7'>12d545e2f9eda489784fc1c4869a70fd50a4f5c7</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32917/perf-test-react-components/Avatar-mount_base.html">609</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32917/perf-test-react-components/Avatar-mount.html">633</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32917/perf-test-react-components/Button-mount_base.html">312</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32917/perf-test-react-components/Button-mount.html">306</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32917/perf-test-react-components/Field-mount_base.html">1142</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32917/perf-test-react-components/Field-mount.html">1132</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32917/perf-test-react-components/FluentProvider-mount_base.html">710</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32917/perf-test-react-components/FluentProvider-mount.html">725</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32917/perf-test-react-components/FluentProviderWithTheme-mount_base.html">85</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32917/perf-test-react-components/FluentProviderWithTheme-mount.html">79</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32917/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">37</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32917/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">37</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32917/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">74</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32917/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">72</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32917/perf-test-react-components/MakeStyles-mount_base.html">866</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32917/perf-test-react-components/MakeStyles-mount.html">885</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32917/perf-test-react-components/Persona-mount_base.html">1788</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32917/perf-test-react-components/Persona-mount.html">1715</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32917/perf-test-react-components/SpinButton-mount_base.html">1405</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32917/perf-test-react-components/SpinButton-mount.html">1395</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SwatchPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32917/perf-test-react-components/SwatchPicker-mount_base.html">1643</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32917/perf-test-react-components/SwatchPicker-mount.html">1679</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

File: change/@fluentui-react-button-87b1b58c-f4f5-4341-b7e9-348961287b4b.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (fabricteam) [inline change/@fluentui-react-button-87b1b58c-f4f5-4341-b7e9-348961287b4b.json:1]: ## 🕵🏾‍♀️ [<b data-uid='vrt-gist'></b> visual regressions to review in the fluentuiv9 Visual Regression Report](https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=32917&commitId=52abf354c98ba385cc50f68906f0fb018eff1559&env=prod&clientType=FLUENTUI)


  <details><summary><b>Avatar Converged</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Avatar Converged.Badge Mask RTL.chromium.png|1| Changed|

</details>


  <details><summary><b>Drawer</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Drawer.overlay drawer full.chromium.png|1162| Changed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

Comment (smhigley) [reply change/@fluentui-react-button-87b1b58c-f4f5-4341-b7e9-348961287b4b.json:1]: no perceivable differences

**Pull request merged**